### PR TITLE
Add x11 service for unclutter

### PR DESCRIPTION
**Pull request by anderspapitto**
modeled after the redshift service

### CODE REVIEW

**Comment by anderspapitto** [comment]
oops, didn't mean to add that second commit

**Comment by lethalman** [comment]
I would be ok with this, but I'd like to understand if there's a way to plug this kind of services into a running X session. Putting it as system service seems wrong to me. What about an xdg autostart? Or what about `displayManager.sessionCommands` ?

**Comment by anderspapitto** [comment]
Well, to me it seems exactly as correct to have this as a service as it does to have redshift as a service (and redshift is already a service) - in both cases, it's a background process that you want running exactly whenever you have an active X session.

Clearly, it could be started in a number of ways, but none of them is quite as general. xdg autostart I believe requires a desktop manager, which I don't currently use. `displayManager.sessionCommands` presumably doesn't work if you turn off the display manager and run `startx` directly (which I've done in the past). Adding a line to `~/.bashrc` won't work if you change your shell to zsh, and so on.

That said I'm not super attached to it, if you really don't want to take the patch that's fine.

File: nixos/modules/services/x11/unclutter.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      type = types.uniq types.string;
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.unclutter = {
+      description = "unclutter";
+      requires = [ "display-manager.service" ];
+      after = [ "display-manager.service" ];
+      wantedBy = [ "graphical.target" ];
```
Comment (lethalman) [inline nixos/modules/services/x11/unclutter.nix]: Not that this service won't start if you rely on graphical.target and xserver is not in autorun (so e.g. doing startx manually somehow).

**Comment by lethalman** [comment]
Merging thanks. I don't like the general path these user services are following... we may solve this problem a day.

**Pull request merged**